orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py

```diff
@@ -0,0 +1,1384 @@
+        else:
+            assert_allclose(value, state_dict_post_checkpoint[key])
+    
+    # load state into pytorch and compare
+    model.load_state_dict(state_dict_pre_checkpoint, strict=False)
+    state_dict_pytorch = model.state_dict()
+    for key, value in state_dict_pytorch.items():
+        assert_allclose(value, state_dict_pre_checkpoint[key])
+
+def test_load_from_single_node_mixed_precision_into_single_node_mixed_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/mixed_precision/'):
+    opts = {
+                'device' : {'id' : device},
+                'mixed_precision':
+                {
+                    'enabled': True,
+                    'loss_scaler': amp.DynamicLossScaler()
+                },
+                'debug' : {'deterministic_compute': True}
+            }
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+
+    state = pickle.load(open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb'))
+    state_dict_pre_checkpoint = state['state_dict']
+
+    # compare all states
+    for key, value in state_dict_pre_checkpoint.items():
+        assert_allclose(value, state_dict_post_checkpoint[key])
+
+def test_load_from_single_node_full_precision_into_single_node_mixed_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/full_precision/'):
+    opts = {
+                'device' : {'id' : device},
+                'mixed_precision':
+                {
+                    'enabled': True,
+                    'loss_scaler': amp.DynamicLossScaler()
+                },
+                'debug' : {'deterministic_compute': True}
+            }
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+
+    state = pickle.load(open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb'))
+    state_dict_pre_checkpoint = state['state_dict']
+
+    # compare all states
+    for key, value in state_dict_post_checkpoint.items():
+        if key.endswith('_fp16'):
+            full_precision_key = key[:-5]
+            assert full_precision_key in state_dict_pre_checkpoint
+            assert_allclose(value, state_dict_pre_checkpoint[full_precision_key], atol=1e-3)
+        else:
+            assert_allclose(value, state_dict_pre_checkpoint[key])
+
+def test_load_from_data_parallelism_full_precision_into_single_node_full_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/data_parallelism/full_precision/'):
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
```

Comment:
I noticed that ` orttrainer.ORTTrainerOptions(opts)` is called for every test. Consider passing just  `opts`  and the method itself instantiates ` ORTTrainerOption`